src/asmcnc/apps/systemTools_app/screen_manager_systemtools.py

```diff
@@ -158,6 +162,16 @@ def get_grbl_settings_from_usb():
     def show_popup_before_download_settings_to_usb(self):
         self.sm.pm.show_download_settings_popup(self)
 
+    def show_popup_before_overwrite_serial_number(self):
+        self.sm.pm.show_overwrite_serial_number_popup(self, button_two_callback=self.call_grbl_overwrite)
+
+    def call_grbl_overwrite(self):
+        grbl_settings_manager = App.get_running_app().grbl_settings_manager
+        if grbl_settings_manager.overwrite_serial_number():
+            self.sm.pm.show_mini_info_popup(self.l.get_str('Serial number overwritten successfully!'))
+        else:
+            self.sm.pm.show_mini_info_popup(self.l.get_str('Could not overwrite serial number!'))
```

Comment:
Do we have a mini_warning_popup as well?

Reply:
No we don't currently